app/javascript/util/log_event.js

```diff
@@ -13,4 +13,51 @@ export const logEvent = (eventName: string, payload: any) => {
   };
 
   if (window.TRACK_USER_ACTIONS) window.mixpanel.track(eventName, opts);
+
+  if (window.ga) logToGa(eventName, payload);
+};
+
+const logToGa = (eventName: string, payload: any) => {
+  const getEventData = () => {
+    switch (eventName) {
+      case 'action:submitted_success':
+        return ['action', 'submitted_success'];
+      case '@@chmp:consent:change_country':
+        return ['gdpr', 'change_country', payload.countryCode];
+      case '@@chmp:consent:change_consent':
+        return ['gdpr', 'change_consent', payload.consented ? 'true' : 'false'];
+      case 'change_amount':
+        return [
+          'fundraising',
+          'change_amount',
+          null,
+          parseFloat(payload.payload),
+        ];
+      case 'set_store_in_vault':
+        return [
+          'fundraising',
+          'set_store_in_valut',
+          payload.payload ? 'true' : 'false',
+        ];
+      case 'set_recurring':
+        return [
+          'fundraising',
+          'set_recurring',
+          payload.payload ? 'true' : 'false',
+        ];
+      case 'fundraiser:transaction_submitted':
+        return [
+          'fundraising',
+          'transaction_submitted',
+          null,
+          parseFloat(payload.amount),
+        ];
+      default:
+        return false;
+    }
+  };
+
+  const eventData = getEventData();
+
+  window.ga('send', 'event', ...eventData);
```

Comment:
Not sure how I feel about having a fn inside of a fn. I'd also rename `payload` to `data` or something similar, to avoid the `payload.payload` confusion/redundancy (as it gives the impression that you are already dealing with the payload, but you're not).

Reply:
👍